src/core/shape_inference/include/scaled_dot_product_attention_shape_inference.hpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include "openvino/op/scaled_dot_product_attention.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v13 {
+template <class T, class TRShape = result_shape_t<T>>
+std::vector<TRShape> shape_infer(const ScaledDotProductAttention* op, const std::vector<T>& input_shapes) {
+    const auto inputs_count = input_shapes.size();
+    const auto has_attention_mask_input = inputs_count >= 4;
+    const auto has_scale_input = inputs_count == 5;
+    NODE_VALIDATION_CHECK(op, inputs_count == 3 || has_attention_mask_input || has_scale_input);
+    TRShape n_dims{TRShape::dynamic()};
+    Dimension s_dim{Dimension::dynamic()};
```

Comment:
Suggestion to be investigated - instead of using single dim type `Dimension`, the following is commonly used instead:
`using DimType = typename T::value_type;`

Reply:
Done